10_rest/willshen8/pkg/puppy/rest_test.go

```diff
@@ -0,0 +1,123 @@
+package puppy
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/go-chi/chi"
+	"github.com/go-chi/chi/middleware"
+	"github.com/stretchr/testify/assert"
+)
+
+func establishRouter() *chi.Mux {
+	// create storer
+	storer := NewMapStore()
+	newPuppy := Puppy{ID: 1, Breed: "Jack Russell Terrier", Color: "White and Brown", Value: "1500"}
```

Comment:
I agree, normally a function should do one thing and does it well (according to the book `clean code`). I relaxed a bit since it's a test file, but will rename to reflect the true purpose.